torch/csrc/utils/tensor_new.cpp

```diff
@@ -436,7 +437,8 @@ Tensor legacy_sparse_tensor_new(c10::TensorTypeId type_id, at::ScalarType scalar
     if (!THPSize_Check(arg) && PyTuple_GET_SIZE(args) >= 1 && arg == PyTuple_GET_ITEM(args, 0)) {
       // new(sequence) binds to this signature but should be treated differently
       // unless the sequences is a torch.Size
-      return legacy_new_from_sequence(type_id, scalar_type, deviceOptional, r.pyobject(0));
+      throw TypeError("SparseTensor.new does not support interpreting Python data as tensor data. " \
```

Comment:
This might not be a very clear error message (I had to read the issue description to understand).
Is the problem that `legacy_new_from_sequence` can only create Tensors with strided layout? Then maybe we want to either:
- Add a layout argument to `legacy_new_from_sequence` to support this properly.
- Change the error message to something like "SparseTensor.new(sequence) only accepts sizes. For sequence containing values, please use the new constructor torch.sparse_coo_tensor() instead.".

Reply:
> Is the problem that legacy_new_from_sequence can only create Tensors with strided layout?

We don't support creating sparse tensors from dense (non-Tensor) data in any constructor.

> Add a layout argument to legacy_new_from_sequence to support this properly.

We shouldn't add functionality to legacy construcotrs.

> Change the error message to something like "SparseTensor.new(sequence) only accepts sizes. For sequence containing values, please use the new constructor torch.sparse_coo_tensor() instead.".

I like this error message better, thanks!

Reply:
ok changed it to:

"SparseTensor.new(sequence) only accepts sizes.  Please use torch.sparse_coo_tensor() or construct a strided tensor and convert it to sparse via to_sparse."